gno/r/launchpad_grc20/token_factory_grc20_test.gno

```diff
@@ -0,0 +1,347 @@
+			if inst.allowBurn != test.expected.allowBurn {
+				t.Errorf("allowBurn = %v, want %v", inst.allowBurn, test.expected.allowBurn)
+			}
+		})
+	}
+}
+
+func TestMint(t *testing.T) {
+	type testMintInput struct {
+		name   string
+		to     std.Address
+		amount uint64
+	}
+
+	type testMintExpected struct {
+		panic       bool
+		totalSupply uint64
+	}
+
+	type testMint struct {
+		input    testMintInput
+		expected testMintExpected
+	}
+
+	type testMintTestTable = map[string]testMint
+
+	bob := std.Address("g126gx6p6d3da4ymef35ury6874j6kys044r7zlg")
+	alice := std.Address("g1ld6uaykyugld4rnm63rcy7vju4zx23lufml3jv")
+
+	std.TestSetOrigCaller(bob)
+
+	NewToken("TestMintTokenMintable", "TestMintTokenMintable", "image", 18, 21_000_000, 23_000_000, true, true)
+	NewToken("TestMintTokenNotMintable", "TestMintTokenNotMintable", "image", 18, 21_000_000, 23_000_000, false, true)
+
+	tests := testMintTestTable{
+		"Success": {
+			input: testMintInput{
+				name:   "TestMintTokenMintable",
+				to:     bob,
+				amount: 1,
+			},
+			expected: testMintExpected{
+				panic:       false,
+				totalSupply: 21_000_001,
+			},
+		},
+		"Token does not exist": {
+			input: testMintInput{
+				name:   "TestToken2",
+				to:     bob,
+				amount: 1000000000000000000,
+			},
+			expected: testMintExpected{
+				panic: true,
+			},
+		},
+		"Minting not allowed": {
+			input: testMintInput{
+				name:   "TestMintTokenNotMintable",
+				to:     bob,
+				amount: 1000000000000000000,
+			},
+			expected: testMintExpected{
+				panic: true,
+			},
+		},
+		"Total supply cap exceeded": {
+			input: testMintInput{
+				name:   "TestMintTokenMintable",
+				to:     bob,
+				amount: 1000000000000000000,
+			},
+			expected: testMintExpected{
+				panic: true,
+			},
+		},
+		"Is not the owner": {
+			input: testMintInput{
+				name:   "TestMintTokenMintable",
+				to:     alice,
+				amount: 1000000000000000000,
+			},
+			expected: testMintExpected{
+				panic: true,
+			},
+		},
+	}
+
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			defer func() {
+				if r := recover(); (r != nil) != test.expected.panic {
+					t.Errorf("panic = %v, want %v", r != nil, test.expected.panic)
+				}
+			}()
+
+			Mint(test.input.name, test.input.to, test.input.amount)
+
+			inst := mustGetToken(test.input.name)
+			if inst.banker.TotalSupply() != test.expected.totalSupply {
+				t.Errorf("totalSupply = %v, want %v", inst.banker.TotalSupply(), test.expected.totalSupply)
+			}
+		})
+	}
+}
+
+func TestBurn(t *testing.T) {
+	type testBurnInput struct {
+		name   string
+		from   std.Address
+		amount uint64
+	}
+
+	type testBurnExpected struct {
+		panic       bool
+		totalSupply uint64
+	}
+
+	type testBurn struct {
+		input    testBurnInput
+		expected testBurnExpected
+	}
+
+	type testBurnTestTable = map[string]testBurn
+
+	bob := std.Address("g126gx6p6d3da4ymef35ury6874j6kys044r7zlg")
+	alice := std.Address("g1ld6uaykyugld4rnm63rcy7vju4zx23lufml3jv")
```

Comment:
Nitpick: you can also use `testutils.TestAddress("alice")`. Might be a bit more readable

Reply:
Yes, thanks for the tip i use it since you told me when we met :rocket: